generator/ServiceClientGeneratorLib/Generators/SourceFiles/ServiceInterfaceNetFramework.tt

```diff
@@ -40,7 +40,10 @@ if (this.Config.ServiceModel.HasPaginators)
     // Generates definitions for each operation in the service model
 		foreach(var operation in this.Config.ServiceModel.Operations)
 		{
-            if(operation.IsInternal || (this.Config.ServiceModel.H2Support == H2SupportDegree.EventStream && operation.IsEventStreamOutput))
+            if(operation.IsInternal || 
```

Comment:
Problem with my original logic as well but we should be checking if `H2Support` is set to `Required` and if so add the `#if NET8_0_OR_GREATER`.